cmd/build

```diff
@@ -36,9 +36,13 @@ fi
 # Builds all docker images
 echo Building docker images
 
-# Build user interface
-echo Building user interface
-if docker build -t test-run/ui -f modules/ui/ui.Dockerfile . ; then
+# Check if UI has already been built (not if -l was used during install)
+if [ ! -d "modules/ui/dist" ]; then
+  cmd/build_ui
+fi
+
+# Build UI image
+if docker build -t testrun/ui -f modules/ui/ui.Dockerfile . ; then
```

Comment:
When not running as root, this appears to block package from working as it has permissions errors on just about every single existing file in the installation directory. Why don't we want this run as root?

Reply:
Found an issue with this PR, looking to change it now.

Reply:
Everything in this PR should now be working. The only thing that you will have to do is clear out the make/usr and make/bin directory in your dev environment. I have also tested installing the new package over an old install and it works.

Reply:
Running package stand alone since it's not inside a controlled environment has no dependency guarantee.  i.e. build_ui needs npm but npm is only installed during the install script, which is not linked to the package method.

Reply:
The idea is that if you are installing via package, then whoever packages testrun will have npm installed (usually myself). If the user is installing from source using cmd/install -l, they should have run cmd/prepare before (which will install npm).

Reply:
Should we add a call to prepare into the package script to make sure the system is always ready?

Reply:
Should automate the removal of existing make directories as well to make this process cleaner.

Reply:
Can't seem to get nmp to install on my system due to various dependency issues.  Should likely move the entire build into a docker container to maintain stability in the build and extract the generated dist from the container.

Reply:
I will move the build process into a container to remove the dependency.